addons/mass_mailing_sms/wizard/mailing_sms_test.py

```diff
@@ -24,16 +24,44 @@ def action_send_sms(self):
         invalid_numbers = [number for number, info in sanitize_res.items() if info['code']]
         if invalid_numbers:
             raise exceptions.UserError(_('Following numbers are not correctly encoded: %s, example : "+32 495 85 85 77, +33 545 55 55 55"', repr(invalid_numbers)))
-        
+
         record = self.env[self.mailing_id.mailing_model_real].search([], limit=1)
         body = self.mailing_id.body_plaintext
         if record:
             # Returns a proper error if there is a syntax error with jinja
             body = self.env['mail.render.mixin']._render_template(body, self.mailing_id.mailing_model_real, record.ids)[record.id]
 
-        self.env['sms.api']._send_sms_batch([{
-            'res_id': 0,
+        sent_sms_list = self.env['sms.api']._send_sms_batch([{
+            'res_id': number,  # used to map the result to the number to log notifications
             'number': number,
             'content': body,
         } for number in sanitized_numbers])
+
+        error_messages = {}
+        if any(sent_sms.get('state') != 'success' for sent_sms in sent_sms_list):
+            # assign an error message to every known error state
+            buy_iap_credits_url = self.sudo().env['iap.account'].get_credits_url(service_name='sms')
+            error_messages = {
```

Comment:
Would move to IAP or SMS so that we may reuse strings.

Reply:
Moved it to 'sms.api ', which makes sense since it contains the method that returns those error states.

Side note: I kept a method returning a dict structure because I want to call "get_credits_url" only if there are actually errors since it induces an RPC if I understood correctly.